docker/docker-compose.etna-app.shared.yml

```diff
@@ -20,14 +20,14 @@ x-base-etna-service: &base-etna-service
   # Base configuration used by most etna services.
   image: etna-base
   volumes: &base-etna-service-volumes
-    - .:/app:cached
+    - .:/app:delegated
```

Comment:
This seems to be an impactful change --- so if we want to test a change to app code, or etna code (as below), does that mean we have to make edits in the container now, not outside on the host?

Reply:
No, not quite.  It's a bit unintuitive at first but think about it:

delegated: the container’s view is authoritative (permit delays before updates on the container appear in the host)

That is only saying that if the container is editing a file, that edit is authoritative (and thus do not pay any sycnhronization cost).  The only places we edit files from the container are from node module, bundler installs, and webpack, and we never edit the outputs of those processes from the host.  Right now mac docker is so slow prceisely because these three processes must pay synchronization costs to build files that are never edited from the host in the first place.

The delay from host -> docker is so small it is not noticeable for individual file edits at the speed of human typing.  It's more noticeable for machine run "download 5000 files as fast as possible".

Reply:
Ah, okay cool, thanks for clarifying! I assume this is what led to issues with npm.